build failed on Fedora

<details>

```bash
[22/37] : && /usr/bin/g++ -fPIC -O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -m64 -march=x86-64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -fPIC -fno-semantic-interposition -fvisibility-inlines-hidden -Werror=date-time -fno-lifetime-dse -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wno-missing-field-initializers -Wimplicit-fallthrough -Wno-nonnull -Wno-class-memaccess -Wno-redundant-move -Wno-pessimizing-move -Wno-noexcept-type -Wdelete-non-virtual-dtor -Wsuggest-override -Wno-comment -Wno-misleading-indentation -Wctad-maybe-unsupported -fdiagnostics-color -ffunction-sections -fdata-sections -fno-common -Woverloaded-virtual -Wcast-qual -fno-strict-aliasing -pedantic -Wno-long-long -Wall -W -Wno-unused-parameter -Wwrite-strings -O2 -g -DNDEBUG  -Wl,-z,relro -Wl,--as-needed  -Wl,-z,pack-relative-relocs -Wl,-z,now -specs=/usr/lib/rpm/redhat/redhat-hardened-ld-errors -specs=/usr/lib/rpm/redhat/redhat-hardened-ld -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -Wl,--build-id=sha1 -specs=/usr/lib/rpm/redhat/redhat-package-notes -Wl,-z,defs -Wl,-z,nodelete   -Wl,-rpath-link,/builddir/build/BUILD/clad-1.3/redhat-linux-build/./lib64  -Wl,--gc-sections -shared -Wl,-soname,libcladPlugin.so.17 -o lib64/libcladPlugin.so.17 tools/CMakeFiles/cladPlugin.dir/ClangPlugin.cpp.o tools/CMakeFiles/cladPlugin.dir/ClangBackendPlugin.cpp.o tools/CMakeFiles/cladPlugin.dir/DerivedFnInfo.cpp.o tools/CMakeFiles/cladPlugin.dir/RequiredSymbols.cpp.o  -Wl,-rpath,"\$ORIGIN/../lib64:"  /usr/lib64/libLLVM-17.so && :
FAILED: lib64/libcladPlugin.so.17 
: && /usr/bin/g++ -fPIC -O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -m64 -march=x86-64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -fPIC -fno-semantic-interposition -fvisibility-inlines-hidden -Werror=date-time -fno-lifetime-dse -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wno-missing-field-initializers -Wimplicit-fallthrough -Wno-nonnull -Wno-class-memaccess -Wno-redundant-move -Wno-pessimizing-move -Wno-noexcept-type -Wdelete-non-virtual-dtor -Wsuggest-override -Wno-comment -Wno-misleading-indentation -Wctad-maybe-unsupported -fdiagnostics-color -ffunction-sections -fdata-sections -fno-common -Woverloaded-virtual -Wcast-qual -fno-strict-aliasing -pedantic -Wno-long-long -Wall -W -Wno-unused-parameter -Wwrite-strings -O2 -g -DNDEBUG  -Wl,-z,relro -Wl,--as-needed  -Wl,-z,pack-relative-relocs -Wl,-z,now -specs=/usr/lib/rpm/redhat/redhat-hardened-ld-errors -specs=/usr/lib/rpm/redhat/redhat-hardened-ld -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -Wl,--build-id=sha1 -specs=/usr/lib/rpm/redhat/redhat-package-notes -Wl,-z,defs -Wl,-z,nodelete   -Wl,-rpath-link,/builddir/build/BUILD/clad-1.3/redhat-linux-build/./lib64  -Wl,--gc-sections -shared -Wl,-soname,libcladPlugin.so.17 -o lib64/libcladPlugin.so.17 tools/CMakeFiles/cladPlugin.dir/ClangPlugin.cpp.o tools/CMakeFiles/cladPlugin.dir/ClangBackendPlugin.cpp.o tools/CMakeFiles/cladPlugin.dir/DerivedFnInfo.cpp.o tools/CMakeFiles/cladPlugin.dir/RequiredSymbols.cpp.o  -Wl,-rpath,"\$ORIGIN/../lib64:"  /usr/lib64/libLLVM-17.so && :
/usr/bin/ld: /tmp/cc1PM7im.ltrans0.ltrans.o: in function `clad::plugin::CladPragmaHandler::HandlePragma(clang::Preprocessor&, clang::PragmaIntroducer, clang::Token&)':
/builddir/build/BUILD/clad-1.3/tools/ClangPlugin.cpp:87:(.text._ZN4clad6plugin17CladPragmaHandler12HandlePragmaERN5clang12PreprocessorENS2_16PragmaIntroducerERNS2_5TokenE+0xf5): undefined reference to `clang::Preprocessor::LexOnOffSwitch(clang::tok::OnOffSwitch&)'
/usr/bin/ld: /tmp/cc1PM7im.ltrans0.ltrans.o: in function `clad::plugin::CladPragmaHandler::HandlePragma(clang::Preprocessor&, clang::PragmaIntroducer, clang::Token&)':
/usr/include/clang/Basic/Diagnostic.h:1320:(.text._ZN4clad6plugin17CladPragmaHandler12HandlePragmaERN5clang12PreprocessorENS2_16PragmaIntroducerERNS2_5TokenE+0xe2): undefined reference to `clang::DiagnosticsEngine::EmitCurrentDiagnostic(bool)'
/usr/bin/ld: /tmp/cc1PM7im.ltrans0.ltrans.o: in function `clad::plugin::CladPlugin::~CladPlugin() [clone .localalias]':
/usr/include/c++/14/bits/unique_ptr.h:93:(.text._ZN4clad6plugin10CladPluginD2Ev+0x1be): undefined reference to `clad::DerivativeBuilder::~DerivativeBuilder()'
/usr/bin/ld: /tmp/cc1PM7im.ltrans0.ltrans.o: in function `clad::plugin::CladPlugin::~CladPlugin() [clone .localalias]':
/usr/include/clang/AST/ASTConsumer.h:43:(.text._ZN4clad6plugin10CladPluginD2Ev+0x1fc): undefined reference to `vtable for clang::ASTConsumer'
/usr/bin/ld: /tmp/cc1PM7im.ltrans0.ltrans.o: in function `llvm::Registry<clang::PragmaHandler>::Add<clad::plugin::CladPragmaHandler>::CtorFn()':
/usr/include/clang/Lex/Pragma.h:70:(.text._ZN4llvm8RegistryIN5clang13PragmaHandlerEE3AddIN4clad6plugin17CladPragmaHandlerEE6CtorFnEv+0x25): undefined reference to `vtable for clang::PragmaHandler'
/usr/bin/ld: /tmp/cc1PM7im.ltrans0.ltrans.o: in function `llvm::Registry<clang::PluginASTAction>::Add<clad::plugin::Action<clad::plugin::CladPlugin> >::CtorFn()':
/usr/include/clang/Frontend/FrontendAction.h:253:(.text._ZN4llvm8RegistryIN5clang15PluginASTActionEE3AddIN4clad6plugin6ActionINS6_10CladPluginEEEE6CtorFnEv+0x2c): undefined reference to `clang::FrontendAction::FrontendAction()'
/usr/bin/ld: /tmp/cc1PM7im.ltrans0.ltrans.o: in function `clad::plugin::Action<clad::plugin::CladPlugin>::~Action()':
/usr/include/clang/Frontend/FrontendAction.h:243:(.text._ZN4clad6plugin6ActionINS0_10CladPluginEED2Ev+0x3d): undefined reference to `vtable for clang::ASTFrontendAction'
/usr/bin/ld: /tmp/cc1PM7im.ltrans0.ltrans.o:/usr/include/clang/Frontend/FrontendAction.h:243:(.text._ZN4clad6plugin6ActionINS0_10CladPluginEED2Ev+0x51): undefined reference to `clang::FrontendAction::~FrontendAction()'
/usr/bin/ld: /tmp/cc1PM7im.ltrans0.ltrans.o: in function `clad::plugin::Action<clad::plugin::CladPlugin>::~Action()':
/usr/include/clang/Frontend/FrontendAction.h:243:(.text._ZN4clad6plugin6ActionINS0_10CladPluginEED0Ev+0x3d): undefined reference to `vtable for clang::ASTFrontendAction'
/usr/bin/ld: /tmp/cc1PM7im.ltrans0.ltrans.o:/usr/include/clang/Frontend/FrontendAction.h:243:(.text._ZN4clad6plugin6ActionINS0_10CladPluginEED0Ev+0x4c): undefined reference to `clang::FrontendAction::~FrontendAction()'
/usr/bin/ld: /tmp/cc1PM7im.ltrans0.ltrans.o: in function `clad::plugin::CladPragmaHandler::~CladPragmaHandler()':
/builddir/build/BUILD/clad-1.3/tools/ClangPlugin.cpp:71:(.text._ZN4clad6plugin17CladPragmaHandlerD2Ev+0xf): undefined reference to `clang::PragmaHandler::~PragmaHandler()'
/usr/bin/ld: /tmp/cc1PM7im.ltrans0.ltrans.o: in function `clad::plugin::CladPragmaHandler::~CladPragmaHandler()':
/builddir/build/BUILD/clad-1.3/tools/ClangPlugin.cpp:71:(.text._ZN4clad6plugin17CladPragmaHandlerD0Ev+0x1b): undefined reference to `clang::PragmaHandler::~PragmaHandler()'
/usr/bin/ld: /tmp/cc1PM7im.ltrans0.ltrans.o: in function `clad::plugin::CladPlugin::CheckBuiltins() [clone .part.0]':
/usr/include/clang/AST/DeclarationName.h:783:(.text._ZN4clad6plugin10CladPlugin13CheckBuiltinsEv.part.0+0xe3): undefined reference to `clang::DeclarationNameLoc::DeclarationNameLoc(clang::DeclarationName)'
/usr/bin/ld: /tmp/cc1PM7im.ltrans0.ltrans.o: in function `clad::plugin::CladPlugin::CheckBuiltins() [clone .part.0]':
/usr/include/clang/Sema/Lookup.h:165:(.text._ZN4clad6plugin10CladPlugin13CheckBuiltinsEv.part.0+0x108): undefined reference to `clang::LookupResult::configure()'
/usr/bin/ld: /tmp/cc1PM7im.ltrans0.ltrans.o: in function `clad::plugin::CladPlugin::CheckBuiltins() [clone .part.0]':
/builddir/build/BUILD/clad-1.3/tools/ClangPlugin.cpp:334:(.text._ZN4clad6plugin10CladPlugin13CheckBuiltinsEv.part.0+0x148): undefined reference to `clang::Sema::LookupQualifiedName(clang::LookupResult&, clang::DeclContext*, bool)'
/usr/bin/ld: /tmp/cc1PM7im.ltrans0.ltrans.o: in function `clad::plugin::CladPlugin::CheckBuiltins() [clone .part.0]':
/usr/include/clang/Sema/Lookup.h:229:(.text._ZN4clad6plugin10CladPlugin13CheckBuiltinsEv.part.0+0x174): undefined reference to `clang::LookupResult::deletePaths(clang::CXXBasePaths*)'
/usr/bin/ld: /usr/include/clang/Sema/Lookup.h:733:(.text._ZN4clad6plugin10CladPlugin13CheckBuiltinsEv.part.0+0x47e): undefined reference to `clang::Sema::CheckLookupAccess(clang::LookupResult const&)'
/usr/bin/ld: /usr/include/clang/Sema/Lookup.h:731:(.text._ZN4clad6plugin10CladPlugin13CheckBuiltinsEv.part.0+0x5c7): undefined reference to `clang::Sema::DiagnoseAmbiguousLookup(clang::LookupResult&)'
/usr/bin/ld: /tmp/cc1PM7im.ltrans0.ltrans.o: in function `clad::checkClangVersion() [clone .localalias]':
/builddir/build/BUILD/clad-1.3/tools/ClangPlugin.cpp:344:(.text._ZN4clad17checkClangVersionEv+0x2b): undefined reference to `clang::getClangFullCPPVersion[abi:cxx11]()'
/usr/bin/ld: /tmp/cc1PM7im.ltrans0.ltrans.o: in function `_GLOBAL__sub_I_ClangPlugin.cpp':
/usr/include/llvm/Support/Registry.h:123:(.text.startup._GLOBAL__sub_I_ClangPlugin.cpp+0x94): undefined reference to `llvm::Registry<clang::PluginASTAction>::add_node(llvm::Registry<clang::PluginASTAction>::node*)'
/usr/bin/ld: /usr/include/llvm/Support/Registry.h:123:(.text.startup._GLOBAL__sub_I_ClangPlugin.cpp+0xf5): undefined reference to `llvm::Registry<clang::PragmaHandler>::add_node(llvm::Registry<clang::PragmaHandler>::node*)'
/usr/bin/ld: /tmp/cc1PM7im.ltrans0.ltrans.o: in function `clad::plugin::CladPlugin::ProcessDiffRequest(clad::DiffRequest&) [clone .localalias]':
/builddir/build/BUILD/clad-1.3/tools/ClangPlugin.cpp:178:(.text._ZN4clad6plugin10CladPlugin18ProcessDiffRequestERNS_11DiffRequestE+0x67): undefined reference to `clang::Sema::PerformPendingInstantiations(bool)'
/usr/bin/ld: /tmp/cc1PM7im.ltrans0.ltrans.o: in function `clad::plugin::CladPlugin::ProcessDiffRequest(clad::DiffRequest&) [clone .localalias]':
/usr/include/clang/AST/Decl.h:2186:(.text._ZN4clad6plugin10CladPlugin18ProcessDiffRequestERNS_11DiffRequestE+0x82): undefined reference to `clang::FunctionDecl::isDefined(clang::FunctionDecl const*&, bool) const'
/usr/bin/ld: /usr/include/clang/AST/Decl.h:2186:(.text._ZN4clad6plugin10CladPlugin18ProcessDiffRequestERNS_11DiffRequestE+0x9d): undefined reference to `clang::FunctionDecl::isDefined(clang::FunctionDecl const*&, bool) const'
/usr/bin/ld: /tmp/cc1PM7im.ltrans0.ltrans.o: in function `clad::plugin::CladPlugin::ProcessDiffRequest(clad::DiffRequest&) [clone .localalias]':
/builddir/build/BUILD/clad-1.3/tools/ClangPlugin.cpp:181:(.text._ZN4clad6plugin10CladPlugin18ProcessDiffRequestERNS_11DiffRequestE+0xd9): undefined reference to `clad::DiffRequest::UpdateDiffParamsInfo(clang::Sema&)'
/usr/bin/ld: /builddir/build/BUILD/clad-1.3/tools/ClangPlugin.cpp:184:(.text._ZN4clad6plugin10CladPlugin18ProcessDiffRequestERNS_11DiffRequestE+0xe8): undefined reference to `clang::LangOptions::LangOptions()'
/usr/bin/ld: /tmp/cc1PM7im.ltrans0.ltrans.o: in function `clad::plugin::CladPlugin::ProcessDiffRequest(clad::DiffRequest&) [clone .localalias]':
/usr/include/clang/AST/Decl.h:290:(.text._ZN4clad6plugin10CladPlugin18ProcessDiffRequestERNS_11DiffRequestE+0x258): undefined reference to `clang::DeclarationName::getAsString[abi:cxx11]() const'
/usr/bin/ld: /tmp/cc1PM7im.ltrans0.ltrans.o: in function `clad::plugin::CladPlugin::ProcessDiffRequest(clad::DiffRequest&) [clone .localalias]':
/builddir/build/BUILD/clad-1.3/tools/ClangPlugin.cpp:241:(.text._ZN4clad6plugin10CladPlugin18ProcessDiffRequestERNS_11DiffRequestE+0x9e7): undefined reference to `clad::DerivativeBuilder::Derive(clad::DiffRequest const&)'
/usr/bin/ld: /builddir/build/BUILD/clad-1.3/tools/ClangPlugin.cpp:210:(.text._ZN4clad6plugin10CladPlugin18ProcessDiffRequestERNS_11DiffRequestE+0xa5e): undefined reference to `llvm::Registry<clad::EstimationPlugin>::begin()'
/usr/bin/ld: /builddir/build/BUILD/clad-1.3/tools/ClangPlugin.cpp:214:(.text._ZN4clad6plugin10CladPlugin18ProcessDiffRequestERNS_11DiffRequestE+0xa9f): undefined reference to `clad::DerivativeBuilder::AddErrorEstimationModel(std::unique_ptr<clad::FPErrorEstimationModel, std::default_delete<clad::FPErrorEstimationModel> >)'
/usr/bin/ld: /builddir/build/BUILD/clad-1.3/tools/ClangPlugin.cpp:194:(.text._ZN4clad6plugin10CladPlugin18ProcessDiffRequestERNS_11DiffRequestE+0xb17): undefined reference to `clang::Decl::dumpColor() const'
/usr/bin/ld: /builddir/build/BUILD/clad-1.3/tools/ClangPlugin.cpp:190:(.text._ZN4clad6plugin10CladPlugin18ProcessDiffRequestERNS_11DiffRequestE+0xb3d): undefined reference to `clang::Decl::print(llvm::raw_ostream&, clang::PrintingPolicy const&, unsigned int, bool) const'
/usr/bin/ld: /tmp/cc1PM7im.ltrans0.ltrans.o: in function `clad::plugin::CladPlugin::ProcessDiffRequest(clad::DiffRequest&) [clone .localalias]':
/usr/include/clang/Basic/Diagnostic.h:869:(.text._ZN4clad6plugin10CladPlugin18ProcessDiffRequestERNS_11DiffRequestE+0xc40): undefined reference to `clang::DiagnosticIDs::getCustomDiagID(clang::DiagnosticIDs::Level, llvm::StringRef)'
/usr/bin/ld: /tmp/cc1PM7im.ltrans0.ltrans.o: in function `clad::plugin::CladPlugin::ProcessDiffRequest(clad::DiffRequest&) [clone .localalias]':
/builddir/build/BUILD/clad-1.3/tools/ClangPlugin.cpp:206:(.text._ZN4clad6plugin10CladPlugin18ProcessDiffRequestERNS_11DiffRequestE+0xc5d): undefined reference to `clang::Sema::Diag(clang::SourceLocation, unsigned int, bool)'
/usr/bin/ld: /builddir/build/BUILD/clad-1.3/tools/ClangPlugin.cpp:209:(.text._ZN4clad6plugin10CladPlugin18ProcessDiffRequestERNS_11DiffRequestE+0xc7c): undefined reference to `clang::Sema::SemaDiagnosticBuilder::~SemaDiagnosticBuilder()'
/usr/bin/ld: /builddir/build/BUILD/clad-1.3/tools/ClangPlugin.cpp:302:(.text._ZN4clad6plugin10CladPlugin18ProcessDiffRequestERNS_11DiffRequestE+0xdb0): undefined reference to `clad::DiffRequest::updateCall(clang::FunctionDecl*, clang::FunctionDecl*, clang::Sema&)'
/usr/bin/ld: /builddir/build/BUILD/clad-1.3/tools/ClangPlugin.cpp:271:(.text._ZN4clad6plugin10CladPlugin18ProcessDiffRequestERNS_11DiffRequestE+0x1114): undefined reference to `clang::Sema::MarkFunctionReferenced(clang::SourceLocation, clang::FunctionDecl*, bool)'
/usr/bin/ld: /builddir/build/BUILD/clad-1.3/tools/ClangPlugin.cpp:273:(.text._ZN4clad6plugin10CladPlugin18ProcessDiffRequestERNS_11DiffRequestE+0x1132): undefined reference to `clang::Sema::MarkFunctionReferenced(clang::SourceLocation, clang::FunctionDecl*, bool)'
/usr/bin/ld: /builddir/build/BUILD/clad-1.3/tools/ClangPlugin.cpp:267:(.text._ZN4clad6plugin10CladPlugin18ProcessDiffRequestERNS_11DiffRequestE+0x1281): undefined reference to `clang::Decl::print(llvm::raw_ostream&, clang::PrintingPolicy const&, unsigned int, bool) const'
/usr/bin/ld: /builddir/build/BUILD/clad-1.3/tools/ClangPlugin.cpp:259:(.text._ZN4clad6plugin10CladPlugin18ProcessDiffRequestERNS_11DiffRequestE+0x12c7): undefined reference to `clang::Decl::dumpColor() const'
/usr/bin/ld: /builddir/build/BUILD/clad-1.3/tools/ClangPlugin.cpp:254:(.text._ZN4clad6plugin10CladPlugin18ProcessDiffRequestERNS_11DiffRequestE+0x12ed): undefined reference to `clang::Decl::print(llvm::raw_ostream&, clang::PrintingPolicy const&, unsigned int, bool) const'
/usr/bin/ld: /tmp/cc1PM7im.ltrans0.ltrans.o: in function `clad::plugin::CladPlugin::HandleTopLevelDecl(clang::DeclGroupRef) [clone .localalias]':
/builddir/build/BUILD/clad-1.3/tools/ClangPlugin.cpp:141:(.text._ZN4clad6plugin10CladPlugin18HandleTopLevelDeclEN5clang12DeclGroupRefE+0xc7): undefined reference to `clad::DiffCollector::DiffCollector(clang::DeclGroupRef, std::vector<clang::SourceRange, std::allocator<clang::SourceRange> >&, llvm::SmallVector<clad::DiffRequest, 16u>&, clang::Sema&)'
/usr/bin/ld: /builddir/build/BUILD/clad-1.3/tools/ClangPlugin.cpp:159:(.text._ZN4clad6plugin10CladPlugin18HandleTopLevelDeclEN5clang12DeclGroupRefE+0x2d8): undefined reference to `clang::Sema::PerformPendingInstantiations(bool)'
/usr/bin/ld: /builddir/build/BUILD/clad-1.3/tools/ClangPlugin.cpp:133:(.text._ZN4clad6plugin10CladPlugin18HandleTopLevelDeclEN5clang12DeclGroupRefE+0x32e): undefined reference to `clad::DerivativeBuilder::DerivativeBuilder(clang::Sema&, clad::plugin::CladPlugin&)'
/usr/bin/ld: /tmp/cc1PM7im.ltrans0.ltrans.o: in function `clad::plugin::CladPlugin::HandleTopLevelDecl(clang::DeclGroupRef) [clone .localalias]':
/usr/include/c++/14/bits/unique_ptr.h:93:(.text._ZN4clad6plugin10CladPlugin18HandleTopLevelDeclEN5clang12DeclGroupRefE+0x347): undefined reference to `clad::DerivativeBuilder::~DerivativeBuilder()'
/usr/bin/ld: /tmp/cc1PM7im.ltrans0.ltrans.o:(.data.rel.ro._ZTIN4clad6plugin17CladPragmaHandlerE+0x10): undefined reference to `typeinfo for clang::PragmaHandler'
/usr/bin/ld: /tmp/cc1PM7im.ltrans0.ltrans.o:(.data.rel.ro._ZTIN4clad6plugin6ActionINS0_10CladPluginEEE+0x10): undefined reference to `typeinfo for clang::PluginASTAction'
/usr/bin/ld: /tmp/cc1PM7im.ltrans0.ltrans.o:(.data.rel.ro._ZTIN4clad6plugin10CladPluginE+0x10): undefined reference to `typeinfo for clang::ASTConsumer'
/usr/bin/ld: /tmp/cc1PM7im.ltrans0.ltrans.o:(.data.rel.ro._ZTVN4clad6plugin6ActionINS0_10CladPluginEEE+0x30): undefined reference to `clang::ASTFrontendAction::ExecuteAction()'
/usr/bin/ld: /tmp/cc1PM7im.ltrans0.ltrans.o:(.data.rel.ro._ZTVN4clad6plugin6ActionINS0_10CladPluginEEE+0x40): undefined reference to `clang::FrontendAction::shouldEraseOutputFiles()'
/usr/bin/ld: /tmp/cc1PM7im.ltrans0.ltrans.o:(.data.rel.ro._ZTVN4clad6plugin6ActionINS0_10CladPluginEEE+0x90): undefined reference to `clang::FrontendAction::EndSourceFile()'
/usr/bin/ld: /tmp/cc1PM7im.ltrans0.ltrans.o:(.data.rel.ro._ZTVN4clad6plugin6ActionINS0_10CladPluginEEE+0x98): undefined reference to `clang::PluginASTAction::anchor()'
/usr/bin/ld: /tmp/cc1PM7im.ltrans0.ltrans.o:(.data.rel.ro._ZTVN4clad6plugin10CladPluginE+0x38): undefined reference to `clang::ASTConsumer::HandleInterestingDecl(clang::DeclGroupRef)'
/usr/bin/ld: /tmp/cc1PM7im.ltrans0.ltrans.o:(.data.rel.ro._ZTVN4clad6plugin10CladPluginE+0x60): undefined reference to `clang::ASTConsumer::HandleTopLevelDeclInObjCContainer(clang::DeclGroupRef)'
/usr/bin/ld: /tmp/cc1PM7im.ltrans0.ltrans.o:(.data.rel.ro._ZTVN4clad6plugin10CladPluginE+0x68): undefined reference to `clang::ASTConsumer::HandleImplicitImportDecl(clang::ImportDecl*)'
collect2: error: ld returned 1 exit status
[23/37] /usr/bin/g++ -DCLAD_INSTDIR_INCL=\"/builddir/build/BUILD/clad-1.3/redhat-linux-build/include\" -DCLAD_SRCDIR_INCL=\"/builddir/build/BUILD/clad-1.3/include\" -D_GNU_SOURCE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I/builddir/build/BUILD/clad-1.3/redhat-linux-build/lib/Differentiator -I/builddir/build/BUILD/clad-1.3/lib/Differentiator -I/builddir/build/BUILD/clad-1.3/include -I/builddir/build/BUILD/clad-1.3/redhat-linux-build/include -O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -m64 -march=x86-64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -fPIC -fno-semantic-interposition -fvisibility-inlines-hidden -Werror=date-time -fno-lifetime-dse -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wno-missing-field-initializers -Wimplicit-fallthrough -Wno-nonnull -Wno-class-memaccess -Wno-redundant-move -Wno-pessimizing-move -Wno-noexcept-type -Wdelete-non-virtual-dtor -Wsuggest-override -Wno-comment -Wno-misleading-indentation -Wctad-maybe-unsupported -fdiagnostics-color -ffunction-sections -fdata-sections -fno-common -Woverloaded-virtual -Wcast-qual -fno-strict-aliasing -pedantic -Wno-long-long -Wall -W -Wno-unused-parameter -Wwrite-strings -O2 -g -DNDEBUG -std=gnu++17 -fPIC   -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS  -fno-exceptions -funwind-tables -MD -MT lib/Differentiator/CMakeFiles/cladDifferentiator.dir/VectorForwardModeVisitor.cpp.o -MF lib/Differentiator/CMakeFiles/cladDifferentiator.dir/VectorForwardModeVisitor.cpp.o.d -o lib/Differentiator/CMakeFiles/cladDifferentiator.dir/VectorForwardModeVisitor.cpp.o -c /builddir/build/BUILD/clad-1.3/lib/Differentiator/VectorForwardModeVisitor.cpp
[24/37] /usr/bin/g++ -DCLAD_INSTDIR_INCL=\"/builddir/build/BUILD/clad-1.3/redhat-linux-build/include\" -DCLAD_SRCDIR_INCL=\"/builddir/build/BUILD/clad-1.3/include\" -D_GNU_SOURCE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I/builddir/build/BUILD/clad-1.3/redhat-linux-build/lib/Differentiator -I/builddir/build/BUILD/clad-1.3/lib/Differentiator -I/builddir/build/BUILD/clad-1.3/include -I/builddir/build/BUILD/clad-1.3/redhat-linux-build/include -O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -m64 -march=x86-64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -fPIC -fno-semantic-interposition -fvisibility-inlines-hidden -Werror=date-time -fno-lifetime-dse -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wno-missing-field-initializers -Wimplicit-fallthrough -Wno-nonnull -Wno-class-memaccess -Wno-redundant-move -Wno-pessimizing-move -Wno-noexcept-type -Wdelete-non-virtual-dtor -Wsuggest-override -Wno-comment -Wno-misleading-indentation -Wctad-maybe-unsupported -fdiagnostics-color -ffunction-sections -fdata-sections -fno-common -Woverloaded-virtual -Wcast-qual -fno-strict-aliasing -pedantic -Wno-long-long -Wall -W -Wno-unused-parameter -Wwrite-strings -O2 -g -DNDEBUG -std=gnu++17 -fPIC   -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS  -fno-exceptions -funwind-tables -MD -MT lib/Differentiator/CMakeFiles/cladDifferentiator.dir/ReverseModeVisitor.cpp.o -MF lib/Differentiator/CMakeFiles/cladDifferentiator.dir/ReverseModeVisitor.cpp.o.d -o lib/Differentiator/CMakeFiles/cladDifferentiator.dir/ReverseModeVisitor.cpp.o -c /builddir/build/BUILD/clad-1.3/lib/Differentiator/ReverseModeVisitor.cpp
[25/37] /usr/bin/g++ -DCLAD_INSTDIR_INCL=\"/builddir/build/BUILD/clad-1.3/redhat-linux-build/include\" -DCLAD_SRCDIR_INCL=\"/builddir/build/BUILD/clad-1.3/include\" -D_GNU_SOURCE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I/builddir/build/BUILD/clad-1.3/redhat-linux-build/lib/Differentiator -I/builddir/build/BUILD/clad-1.3/lib/Differentiator -I/builddir/build/BUILD/clad-1.3/include -I/builddir/build/BUILD/clad-1.3/redhat-linux-build/include -O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -m64 -march=x86-64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -fPIC -fno-semantic-interposition -fvisibility-inlines-hidden -Werror=date-time -fno-lifetime-dse -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wno-missing-field-initializers -Wimplicit-fallthrough -Wno-nonnull -Wno-class-memaccess -Wno-redundant-move -Wno-pessimizing-move -Wno-noexcept-type -Wdelete-non-virtual-dtor -Wsuggest-override -Wno-comment -Wno-misleading-indentation -Wctad-maybe-unsupported -fdiagnostics-color -ffunction-sections -fdata-sections -fno-common -Woverloaded-virtual -Wcast-qual -fno-strict-aliasing -pedantic -Wno-long-long -Wall -W -Wno-unused-parameter -Wwrite-strings -O2 -g -DNDEBUG -std=gnu++17 -fPIC   -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS  -fno-exceptions -funwind-tables -MD -MT lib/Differentiator/CMakeFiles/cladDifferentiator.dir/TBRAnalyzer.cpp.o -MF lib/Differentiator/CMakeFiles/cladDifferentiator.dir/TBRAnalyzer.cpp.o.d -o lib/Differentiator/CMakeFiles/cladDifferentiator.dir/TBRAnalyzer.cpp.o -c /builddir/build/BUILD/clad-1.3/lib/Differentiator/TBRAnalyzer.cpp
[26/37] /usr/bin/g++ -DCLAD_INSTDIR_INCL=\"/builddir/build/BUILD/clad-1.3/redhat-linux-build/include\" -DCLAD_SRCDIR_INCL=\"/builddir/build/BUILD/clad-1.3/include\" -D_GNU_SOURCE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I/builddir/build/BUILD/clad-1.3/redhat-linux-build/lib/Differentiator -I/builddir/build/BUILD/clad-1.3/lib/Differentiator -I/builddir/build/BUILD/clad-1.3/include -I/builddir/build/BUILD/clad-1.3/redhat-linux-build/include -O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -m64 -march=x86-64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -fPIC -fno-semantic-interposition -fvisibility-inlines-hidden -Werror=date-time -fno-lifetime-dse -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wno-missing-field-initializers -Wimplicit-fallthrough -Wno-nonnull -Wno-class-memaccess -Wno-redundant-move -Wno-pessimizing-move -Wno-noexcept-type -Wdelete-non-virtual-dtor -Wsuggest-override -Wno-comment -Wno-misleading-indentation -Wctad-maybe-unsupported -fdiagnostics-color -ffunction-sections -fdata-sections -fno-common -Woverloaded-virtual -Wcast-qual -fno-strict-aliasing -pedantic -Wno-long-long -Wall -W -Wno-unused-parameter -Wwrite-strings -O2 -g -DNDEBUG -std=gnu++17 -fPIC   -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS  -fno-exceptions -funwind-tables -MD -MT lib/Differentiator/CMakeFiles/cladDifferentiator.dir/StmtClone.cpp.o -MF lib/Differentiator/CMakeFiles/cladDifferentiator.dir/StmtClone.cpp.o.d -o lib/Differentiator/CMakeFiles/cladDifferentiator.dir/StmtClone.cpp.o -c /builddir/build/BUILD/clad-1.3/lib/Differentiator/StmtClone.cpp
[27/37] /usr/bin/g++ -DCLAD_INSTDIR_INCL=\"/builddir/build/BUILD/clad-1.3/redhat-linux-build/include\" -DCLAD_SRCDIR_INCL=\"/builddir/build/BUILD/clad-1.3/include\" -D_GNU_SOURCE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I/builddir/build/BUILD/clad-1.3/redhat-linux-build/lib/Differentiator -I/builddir/build/BUILD/clad-1.3/lib/Differentiator -I/builddir/build/BUILD/clad-1.3/include -I/builddir/build/BUILD/clad-1.3/redhat-linux-build/include -O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -m64 -march=x86-64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -fPIC -fno-semantic-interposition -fvisibility-inlines-hidden -Werror=date-time -fno-lifetime-dse -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wno-missing-field-initializers -Wimplicit-fallthrough -Wno-nonnull -Wno-class-memaccess -Wno-redundant-move -Wno-pessimizing-move -Wno-noexcept-type -Wdelete-non-virtual-dtor -Wsuggest-override -Wno-comment -Wno-misleading-indentation -Wctad-maybe-unsupported -fdiagnostics-color -ffunction-sections -fdata-sections -fno-common -Woverloaded-virtual -Wcast-qual -fno-strict-aliasing -pedantic -Wno-long-long -Wall -W -Wno-unused-parameter -Wwrite-strings -O2 -g -DNDEBUG -std=gnu++17 -fPIC   -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS  -fno-exceptions -funwind-tables -MD -MT lib/Differentiator/CMakeFiles/cladDifferentiator.dir/VisitorBase.cpp.o -MF lib/Differentiator/CMakeFiles/cladDifferentiator.dir/VisitorBase.cpp.o.d -o lib/Differentiator/CMakeFiles/cladDifferentiator.dir/VisitorBase.cpp.o -c /builddir/build/BUILD/clad-1.3/lib/Differentiator/VisitorBase.cpp
/builddir/build/BUILD/clad-1.3/lib/Differentiator/VisitorBase.cpp:75: warning: ignoring ‘#pragma clang diagnostic’ [-Wunknown-pragmas]
   75 | #pragma clang diagnostic ignored "-Wunknown-warning-option"
ninja: build stopped: subcommand failed.
```

</details>

build log: https://kojipkgs.fedoraproject.org//work/tasks/9590/113619590/build.log

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -404,6 +404,12 @@ jobs:
             compiler: clang-16
             clang-runtime: '17'
 
+          - name: ubu22-clang16-runtime17-shared-libs
+            os: ubuntu-22.04
+            compiler: clang-16
+            extra_cmake_options: '-DBUILD_SHARED_LIBS=On'
+            clang-runtime: '17'
+
           - name: ubu22-clang16-runtime17-kokkos
             os: ubuntu-22.04
             compiler: clang-16

--- a/lib/Differentiator/CMakeLists.txt
+++ b/lib/Differentiator/CMakeLists.txt
@@ -19,7 +19,8 @@ set_property(SOURCE Version.cpp APPEND PROPERTY
              COMPILE_DEFINITIONS "HAVE_VCS_VERSION_INC")
 
 # (Ab)use llvm facilities for adding libraries.
-add_llvm_library(cladDifferentiator
+llvm_add_library(cladDifferentiator
+  STATIC
   BaseForwardModeVisitor.cpp
   CladUtils.cpp
   ConstantFolder.cpp

--- a/tools/CMakeLists.txt
+++ b/tools/CMakeLists.txt
@@ -8,7 +8,8 @@ set(CLAD_PLUGIN_SRC
   DerivedFnInfo.cpp
   RequiredSymbols.cpp
 )
-add_llvm_library(cladPlugin ${CLAD_PLUGIN_SRC} )
+
+llvm_add_library(cladPlugin STATIC ${CLAD_PLUGIN_SRC})
 
 if (NOT CLAD_BUILD_STATIC_ONLY)
   add_llvm_loadable_module(clad ${CLAD_PLUGIN_SRC} PLUGIN_TOOL clang)

